src/sentry/db/postgres/lookups/array_element_contains.py

```diff
@@ -0,0 +1,32 @@
+from typing import Any
+
+from django.contrib.postgres.fields import ArrayField
+from django.db.models import Lookup
+
+__all__ = ["ArrayElementContains"]
+
+
+class ArrayElementContains(Lookup):
+    lookup_name = "element_contains"
+
+    def as_sql(self, compiler: Any, connection: Any) -> Any:
+        """
+        Custom lookup for checking if an element of the array contains a value.
+        """
+
+        lhs, lhs_params = self.process_lhs(compiler, connection)
+        rhs, rhs_params = self.process_rhs(compiler, connection)
+        params = lhs_params + rhs_params
+
+        return (
+            """EXISTS (
+            SELECT * FROM UNNEST({}) AS elem
+            WHERE elem LIKE '%%' || {} || '%%'
+        )""".format(
+                lhs, rhs
+            ),
```

Comment:
should be an f string

though I'm a little nervous building manual queries using string formatting -- are we guarded against xss here?

Reply:
We are guarded, since we are not inserting the direct value here:
- `lhs` will be column_name <- controlled by us, injection can't happen
- `rhs` will be `'%s'` (always)

`params` that are returned from this `as_sql` function will be the values which are going to be inserted in a query when `cursor.execute` is called, but when this call happens, all parameters inserted in the query are properly escaped ([django code](https://github.com/django/django/blob/cf03aa4e94625971852a09e869f7ee7c328b573f/django/db/models/sql/compiler.py#L1580)).